src/gmail.js

```diff
@@ -1247,6 +1247,13 @@ var Gmail = function(localJQuery) {
         return res;
     };
 
+    api.tools.check_fd_attachment_url = function(url) {
+        var userAccountUrlPart = api.tracker.globals[7];
+        if (url && userAccountUrlPart && url.indexOf(userAccountUrlPart) < 0) {
+            url = url.replace('/mail/?', userAccountUrlPart + '?');
+        }
+    };
+
```

Comment:
You should probably return the ‘url’ variable to the caller too, but otherwise yes, this looks good to me.